### PR TITLE
Sonchau/install docker m1

### PR DESCRIPTION
**Pull request by SonQBChau**
@brouillette The instruction is for local development only, you might want to add your details on the remote server.

### CODE REVIEW

File: docs/install-docker.md [hunk 1]
```diff
@@ -241,3 +241,145 @@ make clean-conda
...
+
+```bash
+sudo nano /etc/hosts
+```
+
+- Add the IP address to the end of the file so it look like this:
+
+```bash
+# Other settings
+xx.xx.xx.xx  docker.localhost
```
Comment (daisieh) [inline docs/install-docker.md]: I thought that you used host.docker.internal?

Comment (SonQBChau) [reply docs/install-docker.md]: yea, I have a note below for local development. If it is confusing I can fix it

Comment (daisieh) [reply docs/install-docker.md]: It just seems to me that if you're using an M1 Mac, you're doing local development.

File: docs/install-docker.md [hunk 2]
```diff
@@ -241,3 +241,145 @@ make clean-conda
...
+cp -i etc/env/example.env .env
+```
+
+- Edit `VENV_OS=arm64mac` for the .env file:
+
+```bash
+# options are [linux, darwin, arm64mac]
+VENV_OS=arm64mac
+VENV_NAME=candig
+```
```
Comment (daisieh) [inline docs/install-docker.md:277]: You should put the host.docker.internal thing for CANDIG_DOMAIN and CANDIG_AUTH_DOMAIN here too.

File: docs/install-docker.md [hunk 3]
```diff
@@ -241,3 +241,145 @@ make clean-conda
...
+```bash
+# Other settings
+xx.xx.xx.xx  docker.localhost
+```
+
+Note: for local development:
+
+```bash
+# Other settings
+127.0.0.1 host.docker.internal
```
Comment (daisieh) [inline docs/install-docker.md:329]: Also do you want 127.0.0.1 for this, or do you want your local IP?

File: docs/install-docker.md (hunk 1 above)
Comment (SonQBChau) [reply docs/install-docker.md]: but Brennan didn't run into this problem. If he didn't have to use host.docker.internal then I will leave the option there

File: docs/install-docker.md (hunk 2 above)
Comment (SonQBChau) [reply docs/install-docker.md:277]: yea I forget. Added

File: docs/install-docker.md (hunk 3 above)
Comment (SonQBChau) [reply docs/install-docker.md:329]: Yea, I want 127.0.0.1 here. my IP didn't work for some reasons

File: docs/install-docker.md (hunk 2 above)
Comment (daisieh) [reply docs/install-docker.md:277]: I don't see the change yet?

File: docs/install-docker.md (hunk 1 above)
Comment (daisieh) [reply docs/install-docker.md]: @brouillette, you were able to use docker.localhost on your M1 setup? Do you know if you had to change any settings anywhere to make that work? @kcranston, what did you use?

File: docs/install-docker.md [hunk 4]
```diff
@@ -241,3 +241,142 @@ make clean-conda
...
+# 2. copy and edit .env with your site's local configuration
+cp -i etc/env/example.env .env
+```
+
+- Edit the .env file:
+
+```bash
+# options are [<ip_addr>, <url>, host.docker.internal, docker.localhost]
+CANDIG_DOMAIN=host.docker.internal
+CANDIG_AUTH_DOMAIN=host.docker.internal
```
Comment (SonQBChau) [inline docs/install-docker.md:272]: @daisieh here is the change in .md

File: docs/install-docker.md (hunk 1 above)
Comment (brouillette) [reply docs/install-docker.md]: Yes

``` 
# .env

...

CANDIG_DOMAIN=docker.localhost
CANDIG_AUTH_DOMAIN=auth.docker.localhost
...
```

Also, adjusting the `/etc/hosts` to point to the machine's local LAN IP helped .. 
Vault wasn't liking trying to resolve `0.0.0.0` or `localhost` from `docker.localhost` or `auth.docker.localhost` 
```
# /etc/hosts

...
# CanDIGv2

192.168.1.XX docker.localhost
192.168.1.XX auth.docker.localhost

...
```

Comment (daisieh) [reply docs/install-docker.md]: So it's still a mystery why Son's machine didn't default to using docker.localhost but used host.docker.internal instead, huh?

**Review by shaikh-rashid — APPROVED** [review]

File: docs/install-docker.md (hunk 1 above)
Comment (SonQBChau) [reply docs/install-docker.md]: this is fixed in #170

**Pull request merged**